1.0.1 fails to build on macosx-10.9-x86_64-3.8 due to kFSEventStreamEventFlagItemCloned

kFSEventStreamEventFlagItemCloned was added in macOS 10.13. Builds for older OS releases should wrap unsupported symbols for backward compatibility even at the loss of minor functionality.

```
    gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/xxx/anaconda3/include -arch x86_64 -I/Users/xxx/anaconda3/include -arch x86_64 -DWATCHDOG_VERSION_STRING="1.0.1" -DWATCHDOG_VERSION_MAJOR=1 -DWATCHDOG_VERSION_MINOR=0 -DWATCHDOG_VERSION_BUILD=1 -I/Users/xxx/proj/macrotomic/play/venv/include -I/Users/xxx/anaconda/include/python3.8 -c src/watchdog_fsevents.c -o build/temp.macosx-10.9-x86_64-3.8/src/watchdog_fsevents.o -std=c99 -pedantic -Wall -Wextra -fPIC -Wno-nullability-completeness -Wno-nullability-extension -Wno-newline-eof -Wno-error=unused-command-line-argument
    src/watchdog_fsevents.c:160:25: error: use of undeclared identifier 'kFSEventStreamEventFlagItemCloned'
    FLAG_PROPERTY(IsCloned, kFSEventStreamEventFlagItemCloned)
                            ^
    src/watchdog_fsevents.c:272:63: warning: unused parameter 'event_ids' [-Wunused-parameter]
                                   const FSEventStreamEventId     event_ids[])
```

--- a/changelog.rst
+++ b/changelog.rst
@@ -9,7 +9,8 @@ Changelog
 202x-xx-xx • `full history <https://github.com/gorakhargosh/watchdog/compare/v1.0.1...master>`__
 
 - [mac] Fix missing ``event_id`` attribute in ``fsevents`` (`#721 <https://github.com/gorakhargosh/watchdog/pull/721>`_)
-- Thanks to our beloved contributors: @SamSchott
+- [mac] Add compatibility with old macOS versions (`#733` <https://github.com/gorakhargosh/watchdog/pull/733>`_)
+- Thanks to our beloved contributors: @SamSchott, @CCP-Aporia
 
 
 1.0.1

--- a/src/watchdog_fsevents.c
+++ b/src/watchdog_fsevents.c
@@ -28,6 +28,18 @@
 
 
 /* Compatibility; since fsevents won't set these on earlier macOS versions the properties will always be False */
+#if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_7
+#error Watchdog module requires at least Mac OS X 10.7
+#endif
+#ifndef MAC_OS_X_VERSION_10_9
+#define MAC_OS_X_VERSION_10_9         1090
+#endif
+#ifndef MAC_OS_X_VERSION_10_10
+#define MAC_OS_X_VERSION_10_10      101000
+#endif
+#ifndef MAC_OS_X_VERSION_10_13
+#define MAC_OS_X_VERSION_10_13      101300
+#endif
 #if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_9
 #define kFSEventStreamEventFlagOwnEvent 0x00080000
 #endif

--- a/src/watchdog_fsevents.c
+++ b/src/watchdog_fsevents.c
@@ -28,14 +28,14 @@
 
 
 /* Compatibility; since fsevents won't set these on earlier macOS versions the properties will always be False */
-#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_9
+#if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_9
 #define kFSEventStreamEventFlagOwnEvent 0x00080000
 #endif
-#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_10
+#if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_10
 #define kFSEventStreamEventFlagItemIsHardlink 0x00100000
 #define kFSEventStreamEventFlagItemIsLastHardlink 0x00200000
 #endif
-#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_13
+#if MAC_OS_X_VERSION_MAX_ALLOWED < MAC_OS_X_VERSION_10_13
 #define kFSEventStreamEventFlagItemCloned 0x00400000
 #endif
 

--- a/src/watchdog_fsevents.c
+++ b/src/watchdog_fsevents.c
@@ -20,10 +20,24 @@
 
 
 #include <Python.h>
+#include <Availability.h>
 #include <CoreFoundation/CoreFoundation.h>
 #include <CoreServices/CoreServices.h>
 #include <stdlib.h>
 #include <signal.h>
+
+
+/* Compatibility; since fsevents won't set these on earlier macOS versions the properties will always be False */
+#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_9
+#define kFSEventStreamEventFlagOwnEvent 0x00080000
+#endif
+#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_10
+#define kFSEventStreamEventFlagItemIsHardlink 0x00100000
+#define kFSEventStreamEventFlagItemIsLastHardlink 0x00200000
+#endif
+#if MAC_OS_X_VERSION_MAX_ALLOWED <= MAC_OS_X_VERSION_10_13
+#define kFSEventStreamEventFlagItemCloned 0x00400000
+#endif
 
 
 /* Convenience macros to make code more readable. */

